include/gsl/pointers

```diff
@@ -267,7 +267,7 @@ public:
     constexpr strict_not_null(const strict_not_null<U>& other) : not_null<T>(other)
     {}
 
-    strict_not_null(strict_not_null&& other) = default;
+    strict_not_null(strict_not_null&& other) noexcept = default;
```

Comment:
```suggestion
    // To avoid invalidating the "not null" invariant, the contained pointer is actually copied
    // instead of moved. If it is a custom pointer, its constructor could in theory throw exceptions.
    strict_not_null(strict_not_null&& other) noexcept(std::is_nothrow_copy_constructible<T>::value) = default;
```

The current behavior is to always copy the contained pointer when invoking this move constructor, to avoid violating the not_null guarantee. For the time being, let's mark this as conditionally no-except, and I'll surface this in the next maintainer's sync.